src/commonMain/kotlin/exchange.dydx.abacus/output/Account.kt

```diff
@@ -2218,22 +2220,37 @@ data class Account(
                 }
             }
 
-            val stakingBalances: IMutableMap<String, AccountBalance> =
+            val stakingBalances = if (staticTyping) {
                 processStakingBalance(
                     existing = existing,
                     parser = parser,
-                    data = data,
                     tokensInfo = tokensInfo,
                     internalState = internalState,
                 )
+            } else {
+                processStakingBalanceDeprecated(
+                    existing = existing,
+                    parser = parser,
+                    data = data,
+                    tokensInfo = tokensInfo,
+                )
+            }
 
-            val stakingDelegations: IMutableList<StakingDelegation> =
+            val stakingDelegations = if (staticTyping) {
```

Comment:
[discussion] the scale of the refactor is pretty enormous. do you think it makes sense to use a separate variable for each piece of state? this would allow us to isolate changes really effectively and be pretty cautious about roll out
cc: @prashanDYDX

Reply:
There is a separate field for each piece of state already.  I will update the params to make it more clear.

The calculation logic operates on multiple fields from the state, so it will be difficult to isolate the changes (e.g., using separate flags).

Reply:
makes sense to me!